rtrlib/rtr_mgr.c

```diff
@@ -331,35 +379,41 @@ int rtr_mgr_init(struct rtr_mgr_config **config_out,
 		goto err;
 	spki_table_init(spki_table, spki_update_fp);
 
-	for (unsigned int i = 0; i < config->len; i++) {
-		struct rtr_mgr_group cg = config->groups[i];
+	config->pfx_table = pfxt;
+	config->spki_table = spki_table;
 
-		cg.status = RTR_MGR_CLOSED;
-		if ((i > 0) && (cg.preference == last_preference)) {
-			MGR_DBG1("Error Same preference for 2 socket groups!");
-			goto err;
-		}
-		if (cg.sockets_len == 0) {
-			MGR_DBG1("Error Empty sockets array in socket group!");
-			goto err;
-		}
-		for (unsigned int j = 0; j < cg.sockets_len; j++) {
-			if (rtr_init(cg.sockets[j], NULL,
+	/* Copy the groups from the array into linked list config->groups */
```

Comment:
I will try to make this a PR on it's own. @mroethke suggested to use the `add_group` function but there might be some refactoring necessary to make this happen.